packages/eas-cli/src/commands/workflow/validate.ts

```diff
@@ -0,0 +1,83 @@
+import fs from 'fs/promises';
+import yaml from 'js-yaml';
+import path from 'path';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import Log from '../../log';
+import { ora } from '../../ora';
+
+export class WorkflowValidate extends EasCommand {
+  static override description = 'validate a workflow configuration yaml file';
+
+  static override args = [
+    {
+      name: 'path',
+      description: 'Path to the workflow configuration YAML file (must end with .yml or .yaml)',
+      required: true,
+    },
+  ];
+
+  async runAsync(): Promise<void> {
+    const {
+      args: { path: filePath },
+    } = await this.parse(WorkflowValidate);
+
+    const spinner = ora().start('Validating the workflow YAML file…');
+
+    try {
+      await validateYAMLAsync(filePath);
+      spinner.succeed('Workflow configuration YAML is valid.');
+    } catch (error) {
+      spinner.fail('Workflow configuration YAML is not valid.');
+
+      if (error instanceof yaml.YAMLException) {
+        Log.error(`YAML syntax error: ${error.message}`);
+      } else if (error instanceof Error) {
+        Log.error(error.message);
+      } else {
+        Log.error(`Unexpected error: ${String(error)}`);
+      }
+
+      throw error;
+    }
+  }
+}
+
+async function checkIfFileExistsAsync(filePath: string): Promise<void> {
+  try {
+    await fs.access(filePath);
+  } catch {
+    throw new Error(`File does not exist: ${filePath}`);
+  }
+}
+
+function checkIfValidYAMLFileExtension(filePath: string): void {
+  const fileExtension = path.extname(filePath).toLowerCase();
+
+  if (fileExtension !== '.yml' && fileExtension !== '.yaml') {
+    throw new Error('File must have a .yml or .yaml extension');
+  }
+}
+
+async function validateYAMLAsync(filePath: string): Promise<void> {
+  await checkIfFileExistsAsync(filePath);
+  checkIfValidYAMLFileExtension(filePath);
+
+  try {
+    const fileContents = await fs.readFile(filePath, 'utf8');
+    const parsedYaml = yaml.load(fileContents);
```

Comment:
It would be much better if the command could also validate the schema, currently only in `www`. I think this is a sensible reason to move the workflow schema to `eas-build-job` later…

Another viable approach would be to use `createWorkflowRun` with some `dry = true` flag, or something. It would _not_ create a workflow run or a workflow revision, only parse and validate with `www`.

Reply:
Yeah, I guess that schema validation would have the biggest benefit for a user here, it would catch all of the typos and so on. Let's move the schema definition to `eas-build-job` and validate against it here.

Reply:
I'm not sure how to move this to `eas-build-job` (not familiar with where that code is or what change to make there). Could we either pair or could one of you help me?

Reply:
>I'm not sure how to move this to eas-build-job (not familiar with where that code is or what change to make there). Could we either pair or could one of you help me?

Don't worry about it @jonsamp, this doesn't need to be a part of this PR and we can do it because we are familiar why how things work there 🙌 

>Also, ideally we'd write a JSON schema that we can validate against, and not only check that the YAML is valid. Adding that where the workflow runs would be great, then exporting it here would be nice, to validate on the server and on the client, and have the same set of rules.

I believe this is what @sjchmiela meant, we have this JSON schema validation implemented using Zod ready and working on both worker-level and WWW-level. We need to move it to `eas-build-job` which is a common lib for such definitions, that is shared between here, WWW, and other places 👍